### PR TITLE
Some janitoring

### LINKED ISSUE #374: std::binary_function is deprecated since g++12
While working on DUNE 2.9 Debian packages I realize, e.g.:
```
[ 66%] Built target tests
In file included from /build/opm-upscaling-2022.10+ds/opm/porsol/common/SimulatorTraits.hpp:43,
                 from /build/opm-upscaling-2022.10+ds/opm/upscaling/UpscalingTraits.hpp:38,
                 from /build/opm-upscaling-2022.10+ds/opm/upscaling/SinglePhaseUpscaler.hpp:39,
                 from /build/opm-upscaling-2022.10+ds/opm/upscaling/RelPermUtils.hpp:33,
                 from /build/opm-upscaling-2022.10+ds/examples/upscale_cap.cpp:67:
/build/opm-upscaling-2022.10+ds/opm/porsol/mimetic/IncompFlowSolverHybrid.hpp:158:35: warning: 'template<class _Arg1, class _Arg2, class _Result> struct std::binary_function' is deprecated [-Wdeprecated-declarations]
  158 |         class axpby : public std::binary_function<T,T,T> {
      |                                   ^~~~~~~~~~~~~~~
In file included from /usr/include/c++/12/bits/stl_tree.h:65,
                 from /usr/include/c++/12/map:60,
                 from /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/mpicxx.h:42,
                 from /usr/lib/x86_64-linux-gnu/openmpi/include/mpi.h:2887,
                 from /usr/include/dune/common/parallel/mpihelper.hh:10,
                 from /build/opm-upscaling-2022.10+ds/examples/upscale_cap.cpp:60:
/usr/include/c++/12/bits/stl_function.h:131:12: note: declared here
  131 |     struct binary_function
      |            ^~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by akva2**
To address clang diagnostics.

Closes #374

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/porsol/mimetic/IncompFlowSolverHybrid.hpp [hunk 1]
```diff
@@ -155,7 +155,7 @@ namespace Opm {
         ///    and usually @code T @endcode is an alias for @code
         ///    double @endcode.
         template<typename T>
-        class axpby : public std::binary_function<T,T,T> {
+        class axpby {
```
Comment (bska) [inline opm/porsol/mimetic/IncompFlowSolverHybrid.hpp:158]: Cool.  Would you mind making the same change to [`MaxAbs`](https://github.com/OPM/opm-upscaling/blob/9219235c5b952452f213068c978203273b322570/opm/core/transport/implicit/NormSupport.hpp#L47), [`SumAbs`](https://github.com/OPM/opm-upscaling/blob/9219235c5b952452f213068c978203273b322570/opm/core/transport/implicit/NormSupport.hpp#L59), and [`Euclid`](https://github.com/OPM/opm-upscaling/blob/9219235c5b952452f213068c978203273b322570/opm/core/transport/implicit/NormSupport.hpp#L71) too?  That way we can close Issue #374 as well.

File: opm/upscaling/UpscalerBase_impl.hpp [hunk 2]
```diff
@@ -296,24 +296,24 @@ namespace Opm
...
 	double side2_area = 0.0;
 
-	int num_faces = 0;
-	int num_bdyfaces = 0;
-	int num_side1 = 0;
-	int num_side2 = 0;
+//	int num_faces = 0;
+//	int num_bdyfaces = 0;
+//	int num_side1 = 0;
+//	int num_side2 = 0;
```
Comment (bska) [inline opm/upscaling/UpscalerBase_impl.hpp:302]: I think we should just remove these, and the associate debugging statement, altogether.  If someone needs the values for debugging/analysis purposes they can make local changes to their copies.

**Review by bska — COMMENTED** [review]
Good start, but we can probably go further in this direction.  Removing `binary_function` is very much needed and should have happened years ago.

File: opm/porsol/mimetic/IncompFlowSolverHybrid.hpp (hunk 1 above)
Comment (akva2) [reply opm/porsol/mimetic/IncompFlowSolverHybrid.hpp:158]: done.

File: opm/upscaling/UpscalerBase_impl.hpp (hunk 2 above)
Comment (akva2) [reply opm/upscaling/UpscalerBase_impl.hpp]: sure. just didn't want to invalidated the commented cout line, but i agree it is better to remove altogether.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge into master now.

**Pull request merged**